core/src/main/java/org/primefaces/extensions/component/keynote/KeynoteRenderer.java

```diff
@@ -0,0 +1,296 @@
+        }
+        else {
+            // static items
+            for (final UIComponent kid : keynote.getChildren()) {
+                if (kid.isRendered()) {
+                    if (kid instanceof UIKeynoteItem) {
+                        // render item
+                        renderItem(context, writer, keynote, (UIKeynoteItem) kid);
+                    }
+                    else {
+                        // render a child like stamped element, etc.
+                        renderChild(context, kid);
+                    }
+                }
+            }
+        }
+
+        writer.endElement("div");
+
+        writer.endElement("div");
+    }
+
+    /**
+     * Create the Javascript.
+     */
+    private void encodeScript(final FacesContext context, final Keynote keynote) throws IOException {
+        final WidgetBuilder wb = getWidgetBuilder(context);
+
+        wb.init("ExtKeynote", keynote)
+                    .attr("width", keynote.getWidth())
+                    .attr("height", keynote.getHeight())
+                    .attr("margin", keynote.getMargin())
+                    .attr("minScale", keynote.getMinScale())
+                    .attr("maxScale", keynote.getMaxScale())
+                    .attr("autoSlide", keynote.getAutoSlide())
+                    .attr("center", keynote.isCenter())
+                    .attr("controls", keynote.isControls())
+                    .attr("disableLayout", keynote.isDisableLayout())
+                    .attr("embedded", keynote.isEmbedded())
+                    .attr("loop", keynote.isLoop())
+                    .attr("navigationMode", keynote.getNavigationMode())
+                    .attr("progress", keynote.isProgress())
+                    .attr("showNotes", keynote.isShowNotes())
+                    .attr("slideNumber", keynote.getSlideNumber())
+                    .attr("touch", keynote.isTouch())
+                    .attr("transition", keynote.getTransition())
+                    .attr("transitionSpeed", keynote.getTransitionSpeed())
+                    .attr("backgroundTransition", keynote.getBackgroundTransition())
+                    .attr("theme", keynote.getTheme());
+
+        if (!LangUtils.isBlank(keynote.getOnslidechanged())) {
+            wb.callback("onslidechanged", "function()", keynote.getOnslidechanged());
+        }
+
+        if (!LangUtils.isBlank(keynote.getOnslidetransitionend())) {
+            wb.callback("onslidetransitionend", "function()", keynote.getOnslidetransitionend());
+        }
+
+        encodeClientBehaviors(context, keynote);
+
+        wb.finish();
+    }
+
+    private void encodeCSS(FacesContext context, String library, String theme) throws IOException {
+        final ResponseWriter writer = context.getResponseWriter();
+        ExternalContext externalContext = context.getExternalContext();
+
+        Resource cssResource = context.getApplication().getResourceHandler().createResource(theme + ".css", library);
```

Comment:
@melloware I remember having issues with loading resources like this. I think we ended up loading them in JavaScript. Do you remember where?

Reply:
Yes I feel like the issue was with inputphone and loading its language files?

Reply:
In that case it must have been the utils script

Reply:
@aripddev can you please try what happens when you Ajax update a section of the page containing a keynote component toggling it from rendered false to true.

Reply:
Hmm. When I added Ajax update targeted to "btnSubmit" in my app, it throws below exception. It was working with PFE showcase.

"Cannot find component for expression "btnSubmit" ....."

Reply:
I'll test it once merged